server/src/main/java/org/elasticsearch/search/vectors/KnnSearchBuilder.java

```diff
@@ -65,55 +69,53 @@ public class KnnSearchBuilder implements Writeable, ToXContentFragment, Rewritea
         } else {
             vectorArray = null;
         }
-        return new KnnSearchBuilder(
-            (String) args[0],
-            vectorArray,
-            (QueryVectorBuilder) args[4],
-            (int) args[2],
-            (int) args[3],
-            (Float) args[5]
-        );
+        return new Builder().field((String) args[0])
+            .queryVector(vectorArray)
+            .queryVectorBuilder((QueryVectorBuilder) args[4])
+            .k((Integer) args[2])
+            .numCandidates((Integer) args[3])
+            .similarity((Float) args[5]);
     });
 
     static {
         PARSER.declareString(constructorArg(), FIELD_FIELD);
         PARSER.declareFloatArray(optionalConstructorArg(), QUERY_VECTOR_FIELD);
-        PARSER.declareInt(constructorArg(), K_FIELD);
-        PARSER.declareInt(constructorArg(), NUM_CANDS_FIELD);
+        PARSER.declareInt(optionalConstructorArg(), K_FIELD);
+        PARSER.declareInt(optionalConstructorArg(), NUM_CANDS_FIELD);
         PARSER.declareNamedObject(
             optionalConstructorArg(),
             (p, c, n) -> p.namedObject(QueryVectorBuilder.class, n, c),
             QUERY_VECTOR_BUILDER_FIELD
         );
         PARSER.declareFloat(optionalConstructorArg(), VECTOR_SIMILARITY);
         PARSER.declareFieldArray(
-            KnnSearchBuilder::addFilterQueries,
+            KnnSearchBuilder.Builder::addFilterQueries,
             (p, c) -> AbstractQueryBuilder.parseTopLevelQuery(p),
             FILTER_FIELD,
             ObjectParser.ValueType.OBJECT_ARRAY
         );
-        PARSER.declareFloat(KnnSearchBuilder::boost, BOOST_FIELD);
+        PARSER.declareFloat(KnnSearchBuilder.Builder::boost, BOOST_FIELD);
         PARSER.declareField(
-            KnnSearchBuilder::innerHit,
+            KnnSearchBuilder.Builder::innerHit,
             (p, c) -> InnerHitBuilder.fromXContent(p),
             INNER_HITS_FIELD,
             ObjectParser.ValueType.OBJECT
         );
     }
 
-    public static KnnSearchBuilder fromXContent(XContentParser parser) throws IOException {
+    public static KnnSearchBuilder.Builder fromXContent(XContentParser parser) throws IOException {
         return PARSER.parse(parser, null);
     }
 
     final String field;
     final float[] queryVector;
     final QueryVectorBuilder queryVectorBuilder;
     private final Supplier<float[]> querySupplier;
-    final int k;
-    final int numCands;
-    final Float similarity;
+    int k;
+    int numCands;
+    Float similarity;
```

Comment:
This can all still be `final` right?

Reply:
Yeap - restored them in 390aff2521de4c95f1a2e7260e7ed5cac6e4f1fa